platform/mbed_watchdog_mgr.cpp

```diff
@@ -0,0 +1,110 @@
+
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifdef DEVICE_WATCHDOG
+
+#include "mbed_watchdog_mgr.h"
+
+static bool is_watchdog_started = false; //boolean to control watchdog start and stop
+#define MS_TO_US(x) ((x) * 1000) //macro to convert millisecond to microsecond
+static uint32_t elapsed_ms = (HW_WATCHDOG_TIMEOUT / 2);
+MBED_STATIC_ASSERT((HW_WATCHDOG_TIMEOUT > 0), "Timeout must be greater than zero");
+
+#if DEVICE_LPTICKER
+/** Create singleton instance of LowPowerTicker for watchdog periodic call back of kick.
+ */
+static SingletonPtr<mbed::LowPowerTicker> _ticker;
+#else
+/** Create singleton instance of Ticker for watchdog periodic call back of kick.
+ */
+static SingletonPtr<mbed::Ticker> _ticker;
+#endif
+
+/** Refreshes the watchdog timer.
+ *
+ * This function should be called periodically before the watchdog times out.
+ * Otherwise, the system is reset.
+ *
+ * If the watchdog timer is not currently running this function does nothing
+ */
+static void mbed_wdog_manager_kick()
+{
+    core_util_critical_section_enter();
```

Comment:
as requested below, is there documentation for this manager? looking at the implementation it kind of mixing hal with driver.

My question is : what is this bringing to the table? why is it not in driver?